builder/transaction_builder.go

```diff
@@ -205,70 +230,73 @@ func (b *TransactionBuilder) AllotRequiredManaAndStoreRemainingManaInOutput(targ
 		}
 
 		// no account bound mana available for the given account, subtract it from the unbounded mana
-		return updateUnboundManaBalance(accountBoundManaOut)
-	}
+		unboundManaInputs, err = safemath.SafeSub(unboundManaInputs, accountBoundManaOut)
+		if err != nil {
+			return ierrors.Wrapf(err, "not enough unbound mana on the input side for account %s", accountID.String())
+		}
 
-	if storedManaOutputIndex >= len(b.transaction.Outputs) {
-		return setBuildError(ierrors.Errorf("given storedManaOutputIndex does not exist: %d", storedManaOutputIndex))
+		return nil
 	}
 
 	// subtract the stored mana on the outputs side
 	for _, o := range b.transaction.Outputs {
 		switch output := o.(type) {
 		case *iotago.AccountOutput:
 			// mana on account outputs is locked to this account
-			if err := updateUnboundAndAccountBoundManaBalances(output.AccountID, output.StoredMana()); err != nil {
-				return setBuildError(ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side"))
+			if err = updateUnboundAndAccountBoundManaBalances(output.AccountID, output.StoredMana()); err != nil {
+				return 0, ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side for account output")
 			}
 
 		default:
 			// check if the output locked mana to a certain account
-			if accountID, isManaLocked := hasManalockCondition(output); isManaLocked {
-				if err := updateUnboundAndAccountBoundManaBalances(accountID, output.StoredMana()); err != nil {
-					return setBuildError(ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side"))
+			if accountID, isManaLocked := b.hasManalockCondition(output); isManaLocked {
+				if err = updateUnboundAndAccountBoundManaBalances(accountID, output.StoredMana()); err != nil {
+					return 0, ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side because of lock")
```

Comment:
```suggestion
					return 0, ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side because of an existing mana lock condition")
```

Reply:
Why is that even a problem with the lock? Do we need to mention that?

Reply:
added just to differentiate from the next error, but its not actually because of, I changed it to "failed to subtract the stored mana on the outputs side, while checking locked mana"